db/compaction_job_test.cc

```diff
@@ -340,6 +340,24 @@ TEST_F(CompactionJobTest, SimpleDeletion) {
   RunCompaction({files}, expected_results);
 }
 
+TEST_F(CompactionJobTest, OutputNothing) {
```

Comment:
sorry but I'm seeing this test case pass both before and after your change, so am not sure it's helping. let me join you in debugging the problem, I finally should have time to look at it.